### PR TITLE
add swav and simclr models to imageclassifier + backbone reorg

### PR DESCRIPTION
**Pull request by teddykoker**
## What does this PR do?

Lets you do:

```python
ImageClassifier(backbone="swav-imagenet")
ImageClassifier(backbone="simclr-imagenet")
```

Also reorgs all the model backbone loading to be in the same place

### CODE REVIEW

File: flash/vision/backbones.py [hunk 1]
```diff
@@ -11,11 +11,71 @@
...
 from typing import Tuple
 
-import torch.nn as nn
 import torchvision
+from pytorch_lightning.utilities import _BOLTS_AVAILABLE
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from torch import nn as nn
+
+if _BOLTS_AVAILABLE:
+    with suppress(TypeError):
```
Comment (carmocca) [inline flash/vision/backbones.py:23]: What is this for? It shouldn'f fail if _BOLTS_AVAILABLE

Comment (carmocca) [reply flash/vision/backbones.py:23]: LGTM otherwise

Comment (kaushikb11) [reply flash/vision/backbones.py:23]: You could remove the `suppress` part.

Comment (teddykoker) [reply flash/vision/backbones.py:23]: Yeah I'm not sure what this was for, it was in the code used to load the bolts models for image embedding

File: flash/vision/backbones.py [hunk 2]
```diff
@@ -11,11 +11,71 @@
...
+        from pl_bolts.models.self_supervised import SimCLR, SwAV
```
Comment (carmocca) [inline flash/vision/backbones.py:24]:
```suggestion
    from pl_bolts.models.self_supervised import SimCLR, SwAV
```

File: flash/vision/backbones.py [hunk 3]
```diff
@@ -11,11 +11,71 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from contextlib import suppress
```
Comment (carmocca) [inline flash/vision/backbones.py:14]:
```suggestion
```

**Review by carmocca — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Pull request merged**